static/app/views/settings/organizationAuth/index.tsx

```diff
@@ -1,114 +1,54 @@
 import type {AuthProvider} from 'sentry/types/auth';
-import type {Organization} from 'sentry/types/organization';
-import withOrganization from 'sentry/utils/withOrganization';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
 
 import OrganizationAuthList from './organizationAuthList';
 
-type Props = DeprecatedAsyncComponent['props'] & {
-  organization: Organization;
-};
-
-type State = DeprecatedAsyncComponent['state'] & {
-  provider: AuthProvider | null;
-  providerList: AuthProvider[] | null;
-};
-
-class OrganizationAuth extends DeprecatedAsyncComponent<Props, State> {
-  componentDidUpdate() {
-    const {organization} = this.props;
-    const access = organization.access;
-
-    if (this.state.provider && access.includes('org:write')) {
-      // If SSO provider is configured, keep showing loading while we redirect
-      // to django configuration view
-      // XXX: This does not need to be normalized for customer-domains because we're going
-      // to a django rendered view.
-      const path = `/organizations/${organization.slug}/auth/configure/`;
-
-      // Use replace so we don't go back to the /settings/auth and hit this path again.
-      window.location.replace(path);
-    }
+function OrganizationAuth() {
+  const organization = useOrganization();
+
+  const {data: providerList, isLoading: loadingProviders} = useApiQuery<AuthProvider[]>(
+    [`/organizations/${organization.slug}/auth-providers/`],
+    {staleTime: 0}
+  );
+  const {data: provider, isLoading: loadingProvider} = useApiQuery<AuthProvider>(
+    [`/organizations/${organization.slug}/auth-provider/`],
+    {staleTime: 0}
+  );
+
+  const shouldRedirectToProvider = provider && organization.access.includes('org:write');
+
+  useEffect(() => {
+    // If SSO provider is configured, keep showing loading while we redirect
+    // to django configuration view
+    // XXX: This does not need to be normalized for customer-domains because we're going
+    // to a django rendered view.
+    const path = `/organizations/${organization.slug}/auth/configure/`;
+
+    // Use replace so we don't go back to the /settings/auth and hit this path again.
+    window.location.replace(path);
+  }, [organization.slug, shouldRedirectToProvider]);
+
+  if (loadingProvider || loadingProviders || shouldRedirectToProvider) {
+    return <LoadingIndicator />;
   }
 
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization} = this.props;
-    return [
-      ['providerList', `/organizations/${organization.slug}/auth-providers/`],
-      ['provider', `/organizations/${organization.slug}/auth-provider/`],
-    ];
+  if (providerList === undefined) {
+    return null;
   }
 
-  /**
-   * TODO(epurkhiser): This does not work right now as we still fallback to the
-   * old SSO auth configuration page
-   */
-  handleConfigure = (provider: AuthProvider) => {
```

Comment:
Just removing these because they were unused and it's very unlikely we need to reference this code in the future